parsers/OPENNEM.py

```diff
@@ -80,181 +81,267 @@ def dataset_to_df(dataset):
-    filtered_datasets = [
-        ds for ds in datasets
-        if ds['type'] == data_type and (
-            (zone_key and ds.get('region') == region)
-            or (sorted_zone_keys and ds.get('id').split('.')[-2] == EXCHANGE_MAPPING_DICTIONARY['->'.join(sorted_zone_keys)]['region_id'])
-        )
-    ]
-    logger.debug('Concatenating datasets..')
+    logger.debug("Parsing JSON..")
+    datasets = r.json()["data"]
+    logger.debug("Filtering datasets..")
+
+    def filter_dataset(ds: dict) -> bool:
+        filter_data_type = ds["type"] == data_type
+        filter_region = False
+        if zone_key:
+            filter_region |= ds.get("region") == region
+        if sorted_zone_keys:
+            filter_region |= (
+                ds.get("id").split(".")[-2]
+                == EXCHANGE_MAPPING_DICTIONARY["->".join(sorted_zone_keys)]["region_id"]
+            )
+        return filter_data_type and filter_region
+
+    filtered_datasets = [ds for ds in datasets if filter_dataset(ds)]
+    logger.debug("Concatenating datasets..")
     df = pd.concat([dataset_to_df(ds) for ds in filtered_datasets], axis=1)
 
     # Sometimes we get twice the columns. In that case, only return the first one
-    is_duplicated_column = df.columns.duplicated(keep='last')
+    is_duplicated_column = df.columns.duplicated(keep="last")
     if is_duplicated_column.sum():
-        logger.warning(f'Dropping columns {df.columns[is_duplicated_column]} that appear more than once')
+        logger.warning(
+            f"Dropping columns {df.columns[is_duplicated_column]} that appear more than once"
+        )
         df = df.loc[:, is_duplicated_column]
 
-    logger.debug('Preparing capacities..')
-    if data_type == 'power' and zone_key:
+    if data_type == "power" and zone_key:
         # SOLAR_ROOFTOP is only given at 30 min interval, so let's interpolate it
-        if 'SOLAR_ROOFTOP' in df:
-            df['SOLAR_ROOFTOP'] = df['SOLAR_ROOFTOP'].interpolate(limit=5)
+        if "SOLAR_ROOFTOP" in df:
+            # TODO decide 5 or 10
+            # I set it to 10 to make sure to minimise impact of NaNs we receive from opennem
+            # https://github.com/opennem/opennem/issues/40
+            df["SOLAR_ROOFTOP"] = df["SOLAR_ROOFTOP"].interpolate(limit=10)
+        logger.debug("Preparing capacities..")
         # Parse capacity data
-        capacities = dict([
-            (obj['id'].split('.')[-2].upper(), obj.get('x_capacity_at_present'))
-            for obj in filtered_datasets if obj['region'] == region
-        ])
+        capacities = dict(
+            [
+                (obj["id"].split(".")[-2].upper(), obj.get("x_capacity_at_present"))
+                for obj in filtered_datasets
+                if obj["region"] == region
+            ]
+        )
         return df, pd.Series(capacities)
     else:
         return df
 
 
-def sum_vector(pd_series, keys, transform=None):
+def sum_vector(pd_series, keys, ignore_nans=False):
```

Comment:
`transform` was dead code. 

Added `ignore_nans` to allow specifying if we choose to allow nans to be counted as 0 or not.